File: IoPTarea1/IoPTarea1/DelegateTableView.m
//
//  DelegateTableView.m
//  IoPTarea1
//
//  Created by Nicolás Gebauer on 22-03-15.
//  Copyright (c) 2015 Nicolás Gebauer. All rights reserved.
//

#import "DelegateTableView.h"
#import "CeldaLetra.h"
#import "CeldaNombre.h"

@implementation DelegateTableView

-(NSInteger) tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    return [_listaAlumnos count];
}

-(UITableViewCell *) tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    NSInteger indexLista = indexPath.row;
    NSString *nombre = [_listaAlumnos objectAtIndex:indexLista];
    if ([nombre length] == 1){
        CeldaLetra *celda = [tableView dequeueReusableCellWithIdentifier:@"IdentificadorCeldaLetra"];
        celda.LabelLetra.text = nombre; //[NSString stringWithFormat:@" %@",nombre];
        [celda setBackgroundColor:[UIColor grayColor]];
        celda.LabelLetra.textColor = [UIColor blackColor];
        return celda;
    }
    else{
        CeldaNombre *celda = [tableView dequeueReusableCellWithIdentifier:@"IdentificadorCeldaNombre"];
        celda.LabelNombre.text = nombre;
        [celda setBackgroundColor:[UIColor whiteColor]];
        return celda;
    }
}

-(void) tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    NSString *nombre = [_listaAlumnos objectAtIndex:indexPath.row];
    if ([nombre length] == 1){
        
    }
    else{
        _numeroCelda = indexPath.row;
        [_referenciaAlControlador performSegueWithIdentifier:@"TransicionMostrarInformacionDetallada" sender:_referenciaAlControlador];
  
    }
}

-(CGFloat) tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath{
    NSInteger indexLista = indexPath.row;
    NSString *nombre = [_listaAlumnos objectAtIndex:indexLista];
    if ([nombre length] == 1){
        return 15;
    }
    else{
        return 40;
    }}

@end

File: IoPTarea2/IoPTarea2/DelegateTableView.swift
//
//  DelegateTableView.swift
//  IoPTarea2
//
//  Created by Nicolás Gebauer on 25-03-15.
//  Copyright (c) 2015 Nicolás Gebauer. All rights reserved.
//

import Foundation
import UIKit

class DelegateTableView : NSObject, UITableViewDelegate, UITableViewDataSource{
    var listaAlumnos : [String] = []
    weak var referenciaAlControlador : UIViewController?
    var numeroCelda : Int = 0
    
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return listaAlumnos.count
    }
    
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let nombre: String = listaAlumnos[indexPath.row]
        if nombre.utf16Count == 1{
            let celdaLetra = tableView.dequeueReusableCellWithIdentifier("IdentificadorCeldaLetra") as CeldaLetra
            celdaLetra.LabelLetra.text = nombre
            celdaLetra.LabelLetra.backgroundColor = UIColor.lightGrayColor()
//            celdaLetra.LabelLetra.textColor = UIColor.lightTextColor()
            return celdaLetra
        }
        else{
            let celdaNombre = tableView.dequeueReusableCellWithIdentifier("IdentificadorCeldaNombre") as CeldaNombre
            celdaNombre.LabelNombre.text = nombre
            return celdaNombre
        }
    }
    
    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
        if referenciaAlControlador != nil{
            let nombre: String = listaAlumnos[indexPath.row]
            if nombre.utf16Count == 1{
                //No hacer nada
            }
            else{
                numeroCelda = indexPath.row
                referenciaAlControlador?.performSegueWithIdentifier("IdentificadorTransicionVistaListaADetalle", sender: referenciaAlControlador)
            }
        }
        tableView.deselectRowAtIndexPath(indexPath, animated: false)
    }
    
    func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
        var nombre: String = listaAlumnos[indexPath.row]
        if nombre.utf16Count == 1{
            return 20
        }
        else{
            return 40
        }
    }
    
    func armarListaBonita() {
        let possibleContent = String(contentsOfFile: "/Users/Nico/Documents/Xcode/IoPTarea2/IoPTarea2/ListaAlumnos.txt", encoding: NSUTF8StringEncoding, error: nil)
        let arrayNombres = possibleContent?.componentsSeparatedByString("\n")
        var listaCompleta : Array<String> = []
        var listaNombres : [String]
        var ultimaLetra = ""
        //Como sacar un trozo de string
        //str.substringWithRange(Range<String.Index>(start: advance(str.startIndex, 2), end: advance(str.endIndex, -1)))
        for nombre in arrayNombres!{
            var nombreOrdenado : String = nombre.capitalizedString
            var primeraLetraSinRevisar = nombreOrdenado.substringToIndex(advance(nombreOrdenado.startIndex, 1))
            var primeraLetraRevisada : String = ""
            switch primeraLetraSinRevisar{
            case "Á":
                primeraLetraRevisada = "A"
            case "É":
                primeraLetraRevisada = "E"
            case "Í":
                primeraLetraRevisada = "I"
            case "Ó":
                primeraLetraRevisada = "O"
            case "Ú":
                primeraLetraRevisada = "U"
            default:
                primeraLetraRevisada = primeraLetraSinRevisar
            }
            if primeraLetraRevisada != ultimaLetra{
                ultimaLetra = primeraLetraRevisada
                listaCompleta.append(ultimaLetra)
            }
            listaCompleta.append(nombreOrdenado)
        }
        listaAlumnos = listaCompleta
    }
}